src/dos/drive_fat.cpp

```diff
@@ -1075,8 +1075,7 @@ bool fatDrive::FileCreate(DOS_File **file, char *name, uint16_t attributes) {
                                     fileEntry.loFirstClust,
                                     fileEntry.entrysize,
                                     this);
-	bool is_readonly     = fileEntry.attrib & DOS_ATTR_READ_ONLY;
-	fat_file->flags      = is_readonly ? OPEN_READ : OPEN_READWRITE;
+	fat_file->flags      = OPEN_READWRITE;
```

Comment:
What is the purpose of this change?

Reply:
I was under false impression that creating a read-only file on DOS returns a handle that you can't write to but testing showed that I was wrong. You *can* write to a freshly created r/o file until you close the handle. I.e. r/o attribute merely blocks you from opening new r/w streams, it doesn't block existing r/w streams from working.

Reply:
Ok, makes sense.